mix/entry_connection.nim

```diff
@@ -1,22 +1,20 @@
-import hashes, chronos, stew/byteutils
+import hashes, chronos, std/options, stew/byteutils
 import libp2p/stream/connection
 import ./protocol
 
 type MixDialer* = proc(
-  msg: seq[byte], proto: ProtocolType, destMultiAddr: MultiAddress, destPeerId: PeerId
+  msg: seq[byte],
+  proto: ProtocolType,
+  destMultiAddr: Option[MultiAddress],
+  destPeerId: PeerId,
 ): Future[void] {.async: (raises: [CancelledError, LPStreamError], raw: true).}
 
 type MixEntryConnection* = ref object of Connection
-  destMultiAddr: MultiAddress
+  destMultiAddr: Option[MultiAddress]
```

Comment:
@AkshayaMani this change here. I suspect this comes down to some sort of subtle difference between the gs poc and the waku approach. does this need some sort of careful consideration, and might it allude to some important impl details?